### PR TITLE
Fix/self service

### PR DESCRIPTION
**Pull request by hirsch88**

### CODE REVIEW

**Comment by incubator-tekton** [comment]
:rocket: Welcome to Baloise ChatOps based GitOps Bot :tada:
                                
    - If you want to enable preview environments for this PullRequest, comment with `/preview`. 
    - If you want to disable the preview environment for this PR, comment with `/undeploy`. 
    - If you need help or want to see the docs, comment with `/help`. 
                                
    Make sure to refresh this page after submitting a command.
    Preview environments can be redeployed automatically if you enable the webhook for updates to the source branch. :new:
    
    Have fun building awesome apps! :hammer:  :whale2:  :truck:  :ship:

File: packages/components-angular/src/overlays/overlay.ts [hunk 1]
```diff
@@ -21,6 +22,13 @@ export class OverlayBaseController<Opts, Overlay> implements ControllerShape<Opt
     return this.ctrl.dismiss(data, role, id)
   }
 
+  /**
+   * When `id` is not provided, it dismisses the top overlay.
```
Comment (svene) [inline packages/components-angular/src/overlays/overlay.ts]: adapt comment

File: packages/components/src/components/form/bal-select/bal-select.tsx [hunk 2]
```diff
@@ -384,11 +396,16 @@ export class Select {
...
   private get optionArray() {
-    return Array.from(this.options, ([_, value]) => value).filter(option => {
-      if (this.typeahead) {
+    const options = Array.from(this.options, ([_, value]) => value)
+    if (!this.typeahead || this.remote) {
+      return options
+    }
+
+    return options.filter(option => {
+      if (this.filter === 'includes') {
```
Comment (svene) [inline packages/components/src/components/form/bal-select/bal-select.tsx]: could be changed to:
  return (this.filter === 'includes') ? includes(...) : startsWith(...)

File: packages/components-angular/src/overlays/overlay.ts (hunk 1 above)
Comment (hirsch88) [reply packages/components-angular/src/overlays/overlay.ts]: done thanks

File: packages/components/src/components/form/bal-select/bal-select.tsx (hunk 2 above)
Comment (hirsch88) [reply packages/components/src/components/form/bal-select/bal-select.tsx]: done thanks

**Pull request merged**